python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -74,6 +73,7 @@ class Node(object):
     thread-safe usage.
     """
     __metaclass__ = abc.ABCMeta
+    __slots__ = ('_index_cache_dict', '_parent', '_read_lock', '_write_lock', '_dirty', '_registry', '_data_dict')
```

Comment:
@milliams is right though that we shouldn't add `__dict__` back into `__slots__` for a generic base class as this undoes most of the benefit. If you want the ability to have dynamic variables then just don't use `__slots__` or possibly add the `__dict__` back **only** for classes that you want to allow them. That way the other classes benefit from the `__slots__`.

I would be keen to know however where we are using dynamically assigned variables and would argue that we should perhaps review if they are necessary.